src/main.js

```diff
@@ -729,21 +729,29 @@ class MiniGraphCard extends LitElement {
 
   updateBounds({ config } = this) {
     this.bound = [
-      config.lower_bound !== undefined
-        ? config.lower_bound
-        : Math.min(...this.primaryYaxisSeries.map(ele => ele.min)) || this.bound[0],
-      config.upper_bound !== undefined
-        ? config.upper_bound
-        : Math.max(...this.primaryYaxisSeries.map(ele => ele.max)) || this.bound[1],
+      config.lower_bound === undefined
+        ? Math.min(...this.primaryYaxisSeries.map(ele => ele.min)) || this.bound[0]
+        : config.soft_bounds
+          ? Math.min(config.lower_bound, ...this.primaryYaxisSeries.map(ele => ele.min))
+          : config.lower_bound,
+      config.upper_bound === undefined
+        ? Math.max(...this.primaryYaxisSeries.map(ele => ele.max)) || this.bound[1]
+        : config.soft_bounds
+          ? Math.max(config.upper_bound, ...this.primaryYaxisSeries.map(ele => ele.max))
+          : config.upper_bound,
     ];
 
     this.boundSecondary = [
-      config.lower_bound_secondary !== undefined
-        ? config.lower_bound_secondary
-        : Math.min(...this.secondaryYaxisSeries.map(ele => ele.min)) || this.boundSecondary[0],
-      config.upper_bound_secondary !== undefined
-        ? config.upper_bound_secondary
-        : Math.max(...this.secondaryYaxisSeries.map(ele => ele.max)) || this.boundSecondary[1],
+      config.lower_bound_secondary === undefined
+        ? Math.min(...this.secondaryYaxisSeries.map(ele => ele.min)) || this.boundSecondary[0]
+        : config.soft_bounds_secondary
+          ? Math.min(config.lower_bound_secondary, ...this.secondaryYaxisSeries.map(ele => ele.min))
+          : config.lower_bound_secondary,
+      config.upper_bound_secondary === undefined
+        ? Math.max(...this.secondaryYaxisSeries.map(ele => ele.max)) || this.boundSecondary[1]
+        : config.soft_bounds_secondary
+          ? Math.max(config.upper_bound_secondary, ...this.secondaryYaxisSeries.map(ele => ele.max))
+          : config.upper_bound_secondary,
```

Comment:
Sorry but I sustain my opinion: nested inline conditions make code not clear and hard to read. Please change it to regular `if` statements.

Another idea came up to my mind now: what about not adding new setting entry but allow values for bounds like: `~0`. The `~` prefix would mean that this is not hard boundary . What do you think?

I would do it in the following way:
```js
this.bound = [
    this.getBoudary("min", config.lower_bound, this.primaryYaxisSeries, this.bound[0]),
    this.getBoudary("max", config.upper_bound, this.primaryYaxisSeries, this.bound[1]),
];
```

```js
getBoudary(type, configVal, series, defaultVal) {
    if (configVal === undefined) {
        // dynamic boundary depending on values
        return Math[type](...series.map(ele => ele[type])) || defaultVal
    }

    if (configVal[0] != "~") {
        // fixed boundary
        return configVal;
    }

    // soft boundary (respecting out of range values)
    configVal = parseInt(configVal.substr(1));
    Math[type](configVal, ...series.map(ele => ele[type]));
}
```

Reply:
Good idea. Now config looks much better: [Screenshot](https://s.mail.ru/KW4e/tG6Cpzzdy)